cpp/src/io/csv/csv_reader.cu

```diff
@@ -577,53 +579,61 @@ gdf_error read_csv(csv_read_arg *args)
 	assert(h_uncomp_data != nullptr);
 	assert(h_uncomp_size != 0);
 
-	error = countRecordsAndQuotes(h_uncomp_data, h_uncomp_size, raw_csv);
+	error = countRecordsAndQuotes(h_uncomp_data, h_uncomp_size, &raw_csv);
 	checkError(error, "call to count the number of rows");
 
-	error = setRecordStarts(h_uncomp_data, h_uncomp_size, raw_csv);
+	error = setRecordStarts(h_uncomp_data, h_uncomp_size, &raw_csv);
 	checkError(error, "call to store the row offsets");
 
-	error = uploadDataToDevice(h_uncomp_data, h_uncomp_size, raw_csv);
+	error = uploadDataToDevice(h_uncomp_data, h_uncomp_size, &raw_csv);
 	checkError(error, "call to upload the CSV data to the device");
 
+	//-----------------------------------------------------------------------------
+	//---  done with host data
+	if (args->input_data_form == gdf_csv_input_form::FILE_PATH)
+	{
+		close(fd);
+		munmap(map_data, map_size);
+	}
+
 	//-----------------------------------------------------------------------------
 	//-- Populate the header
 
 	// Check if the user gave us a list of column names
 	if(args->names == nullptr) {
 
-		error = setColumnNamesFromCsv(raw_csv);
+		error = setColumnNamesFromCsv(&raw_csv);
 		if (error != GDF_SUCCESS) {
 			return error;
 		}
-		const int h_num_cols = raw_csv->col_names.size();
+		const int h_num_cols = raw_csv.col_names.size();
 
 		// Allocating a boolean array that will use to state if a column needs to read or filtered.
-		raw_csv->h_parseCol = (bool*)malloc(sizeof(bool) * (h_num_cols));
-		RMM_TRY( RMM_ALLOC((void**)&raw_csv->d_parseCol,(sizeof(bool) * (h_num_cols)),0 ) );
+		raw_csv.h_parseCol = std::make_unique<bool[]>(h_num_cols);
+		raw_csv.d_parseCol = device_buffer<bool>(h_num_cols);
```

Comment:
I'd prefer using `thrust::host_vector` and `rmm::device_vector` for `h_parseCol` and `d_parseCol`, respectively, since we are using it like a vector in the code.

We can simply assign `d_parseCol = h_parseCol` and not require explicit device_buffer<bool> upfront here.

Reply:
I agree, just forgot due to inertia. I'll make this change.

Reply:
Done. Pretty happy with the effect.